crates/bevy_ecs/src/system/query.rs

```diff
@@ -1546,11 +1522,34 @@ impl<'w, 's, D: QueryData, F: QueryFilter> Query<'w, 's, D, F> {
     /// - [`get_many`](Self::get_many) to get read-only query items without checking for duplicate entities.
     /// - [`many_mut`](Self::many_mut) for the panicking version.
     #[inline]
-    pub fn get_many_mut<const N: usize>(
+    pub fn many_mut<const N: usize>(
         &mut self,
         entities: [Entity; N],
     ) -> Result<[D::Item<'_>; N], QueryEntityError> {
-        self.reborrow().get_many_inner(entities)
+        self.reborrow().many_inner(entities)
+    }
+
+    /// Returns the query items for the given array of [`Entity`].
+    /// This consumes the [`Query`] to return results with the actual "inner" world lifetime.
+    ///
+    /// Deprecated alias for [`many_inner`](Self::many_inner).
+    #[inline]
+    #[deprecated(note = "Use `many_inner` instead, which now returns a Result.")]
+    pub fn get_many_inner<const N: usize>(
+        self,
+        entities: [Entity; N],
+    ) -> Result<[D::Item<'w>; N], QueryEntityError<'w>> {
+        // Verify that all entities are unique
+        for i in 0..N {
```

Comment:
Could you do something like this if you wanted to? It might be easier to understand. Totally a nit though

```
if len(x) > len(set(x)):
```

Reply:
Probably, but I've opted to not change the logic in this PR. Despite the diff, this was just a pair of renames.